test/e2e/tests/confirmations/transactions/erc20-approve-redesign.spec.ts

```diff
@@ -116,55 +113,33 @@ async function createERC20ApproveTransaction(driver: Driver) {
 async function assertApproveDetails(driver: Driver) {
   await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
 
-  await driver.waitForSelector({
-    css: 'h2',
-    text: 'Spending cap request',
-  });
-
-  await driver.waitForSelector({
-    css: 'p',
-    text: 'This site wants permission to withdraw your tokens',
-  });
-
-  await driver.waitForSelector({
-    css: 'p',
-    text: 'Estimated changes',
-  });
-
-  await driver.waitForSelector({
-    css: 'p',
-    text: 'Spending cap',
-  });
-
-  await driver.waitForSelector({
-    css: 'p',
-    text: '7',
-  });
+  const erc20ApproveConfirmation = new ERC20ApproveTransactionConfirmation(
+    driver,
+  );
 
-  await toggleAdvancedDetails(driver);
+  await erc20ApproveConfirmation.checkSpendingCapRequestTitle();
+  await erc20ApproveConfirmation.checkSpendingCapPermissionDescription();
+  await erc20ApproveConfirmation.checkEstimatedChangesSection();
+  await erc20ApproveConfirmation.checkSpendingCapSection();
+  await erc20ApproveConfirmation.checkSpendingCapAmount('7');
 
-  await driver.waitForSelector({
-    css: 'p',
-    text: 'Spender',
-  });
+  // TODO - Scroll button not shown in Firefox if advanced details enabled too fast.
+  await driver.delay(1000);
+  await erc20ApproveConfirmation.clickAdvancedDetailsButton();
 
-  await driver.waitForSelector({
-    css: 'p',
-    text: 'Request from',
-  });
+  await erc20ApproveConfirmation.checkAdvancedDetailsSections();
+  await erc20ApproveConfirmation.checkSpendingCapSection();
+}
 
-  await driver.waitForSelector({
-    css: 'p',
-    text: 'Interacting with',
-  });
+async function confirmApproveTransaction(driver: Driver) {
```

Comment:
in general, I think it's better to avoid functions like this in the spec files (encapsulating test steps on a function inside a spec file). I think it makes the debugging experience a bit worse, as well as doesn't fully follow POM --> if we happen to see this same steps over an over, we should create a flow file. However, this is only used in this spec, so I would rather remove the function and add the steps on each spec, otherwise might be a bit of an overkill.
What do you think?